app/client-app/src/app/pages/Editor/components/PageList/PageHeader.js

```diff
@@ -1,12 +1,33 @@
-import { Flex, useColorMode, Icon } from "@chakra-ui/react";
+import { Flex, useColorMode, Icon, Input } from "@chakra-ui/react";
+import { useState, useEffect, useRef } from "react";
 import LightHeading from "components/core/LightHeading";
 import { FaGripVertical } from "react-icons/fa";
 import PageActionButtons from "./PageActionButtons";
 import { defaultColorMode } from "themes";
-
+import axios from "axios";
+import { useFetchSurvey } from "app/contexts/FetchSurvey";
+import { authorization_BearerToken, withHeaders, contentType_AppJson } from "api/helpers";
 const PageHeader = ({ page, order, dragHandleProps }) => {
   const { colorMode } = useColorMode();
   const headerStyle = { light: { bg: "gray.300" }, dark: { bg: "gray.600" } };
+  const { id, mutate } = useFetchSurvey();
+  const [name, setName] = useState(page.name);
+  const firstUpdate = useRef(true);
+  useEffect(() => {
```

Comment:
don't need the `useRef`/`useEffect` all in the main component body. At the very least move to a custom hook, although for this purpose we already have one that e.g. Survey NameInput uses.

- [hook to run a function after a resettable timeout](https://github.com/decsys/decsys/blob/810ace8e54993aa0f4d4858b6eacaf80594ffa0e/app/client-app/src/hooks/useDeferredAction.js)
- [`NameInput` using it](https://github.com/decsys/decsys/blob/810ace8e54993aa0f4d4858b6eacaf80594ffa0e/app/client-app/src/components/shared/NameInput.js#L19)